### PR TITLE
Add endpoint to fetch Elasticsearch view mappings

### LINKED ISSUE #4142: Add an endpoint to return the current mapping of an Elasticsearch view
**Motivation**
Users can define a mapping in Elasticsearch views.
However if a field is not defined in this mapping, Elasticsearch can deduce it via dynamic mapping, sometimes badly as the first appearance of the field define the field type for the other appearances

**Acceptance criteria**
1. Users need the `views/write` permission
2. When a user calls the endpoint, the mapping is fetched to the user and passed on to the client

### PR DESCRIPTION
**Pull request by olivergrabinski**
New pass-through endpoint to ES mappings based on user's authorization

Closes #4142

### CODE REVIEW

File: delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/model/ElasticSearchViewRejection.scala [hunk 1]
```diff
@@ -126,7 +126,7 @@ object ElasticSearchViewRejection {
     *   the view id
     */
   final case class DifferentElasticSearchViewType(
-      id: Iri,
+      id: String,
```
Comment (olivergrabinski) [inline delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/model/ElasticSearchViewRejection.scala:129]: To be able to pass an IdSegment as well, thought the segment might not be fully expanded 🤔

File: tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/ElasticSearchViewsSpec.scala [hunk 2]
```diff
@@ -421,5 +421,19 @@ class ElasticSearchViewsSpec extends BaseSpec with EitherValuable with CirceEq {
...
+
+        def hasOnlyOneKey = (j: ACursor) => j.keys.exists(_.size == 1)
+        def downFirstKey  = (j: ACursor) => j.downField(j.keys.get.head)
+
+        assert(hasOnlyOneKey(json.hcursor))
+        val firstKey = downFirstKey(json.hcursor)
+        assert(hasOnlyOneKey(firstKey))
+        assert(downFirstKey(firstKey).key.contains("mappings"))
+      }
+    }
```
Comment (olivergrabinski) [inline tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/ElasticSearchViewsSpec.scala:436]: Not sure what checks beyond the basic structure make sense

File: delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/ElasticSearchViewsQuery.scala [hunk 3]
```diff
@@ -94,6 +102,29 @@ final class ElasticSearchViewsQueryImpl private[elasticsearch] (
     } yield search
   }
 
+  override def mapping(
+      id: IdSegment,
+      project: ProjectRef
+  )(implicit caller: Caller): IO[ElasticSearchViewRejection, Json] =
+    for {
+      view   <- aclCheck.authorizeForOr(project, permissions.write)(AuthorizationFailed).as(viewStore.fetch(id, project))
```
Comment (olivergrabinski) [inline delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/ElasticSearchViewsQuery.scala:110]: Authorization done in a similar fashion to the `query` functions

File: tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/ElasticSearchViewsSpec.scala [hunk 4]
```diff
@@ -421,5 +421,19 @@ class ElasticSearchViewsSpec extends BaseSpec with EitherValuable with CirceEq {
       }
     }
 
+    "return the view's mapping" in {
```
Comment (imsdu) [inline tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/ElasticSearchViewsSpec.scala:424]: We need other tests with a user with a missing permission and for a non-existing view

**Review by imsdu — CHANGES_REQUESTED** [review]

**Review by imsdu — APPROVED** [review]

**Pull request merged**